### PR TITLE
Add Forgejo

### PR DESCRIPTION
**Pull request by caesar**
Thanks for taking the time to suggest an addition to awesome-selfhosted! 

To ensure your Pull Request is dealt with swiftly, please check the following (check the boxes `[x]`):
- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [X] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [x] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The file you are adding is formatted as described in [addition.md](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/ISSUE_TEMPLATE/addition.md).
- [x] Comments and unused optional fields have been removed.
- [x] The file you are adding uses [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) file naming, for example `my-awesome-software.yml`.
- [x] Values for `platform` are the main **server-side** requirements for the software. Don't include frameworks or specific dialects.
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding was first released more than 4 months ago.
- [x] Any software project you are adding has working installation instructions.
- [x] If you are adding software forked from another active project, please provide/link to a clear list of differences between both. _-> Primary difference is governance, but see https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/91#issuecomment-1694422059 for more details and more differences_

### CODE REVIEW

File: software/forgejo.yml
Comment (KieranRobson) [inline software/forgejo.yml:8]: > https://forgejo.org/docs/latest/admin/installation/

As Docker is first, its listed first.

```suggestion
platforms:
 - Docker
   - Go
```

**Review by KieranRobson — CHANGES_REQUESTED** [review]
Quick change to be made.

File: software/forgejo.yml
Comment (caesar) [reply software/forgejo.yml]: Thanks for the suggestion @KieranRobson, I've made that change.

**Review by kokomo123 — APPROVED** [review]
Seeing the differences now of forgejo, i can confidently say that I approve this PR as it's a good license, it's stable, it's now different from Gitea, and there's no issues with this. Thanks for your contribution 👍🏻

**Comment by nodiscc** [comment]
Hi @caesar any news about the information requested in https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3467?

I forgot to port the guideline that was added in https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3998/files to the new repo (`If you are adding software forked from another project, please provide/link to a clear list of differences between both.`), but I think it would be good to have.

Good examples here: https://docs.gitea.com/features/comparison, https://about.gitlab.com/competition/github/, etc.

Also the fork relationship with Gitea should be mentioned as we do for other projects: https://github.com/search?q=repo%3Aawesome-selfhosted%2Fawesome-selfhosted-data%20%22fork%20of%22&type=code

**Comment by kokomo123** [comment]
https://forgejo.org/docs/v1.20/user/blocking-user/, per the comment before yours, he noted some moderation tools that weren't in Gitea, also he noted some changes with a big comment, per https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3467#issuecomment-1636844524 @nodiscc

**Review by kokomo123 — COMMENTED** [review]
Per my comment above, I will be changing this back to "general" feedback instead of a full approval.

**Comment by nodiscc** [comment]
> I forgot before reviewing that we must include the fork relationship key

It is not written down anywhere, so no worries, however I think it would be good to have a `(fork of ...)` mention for projects that are forks of other, active projects (similar to what was done in https://github.com/awesome-selfhosted/awesome-selfhosted/commit/ecd7932cdf0a3c3d24bd1d10108f33d556fb74e0).

> and the `If you are adding software forked from another project...`

Again, I forgot to re-add this guideline after migrating to the new repo, so no worries. I will re-add it now.

**Comment by kokomo123** [comment]
Speaking of which in a off-topic related note with this PR, can you also add a "on hold" label for PRs?

**Comment by nodiscc** [comment]
> he noted some moderation tools that weren't in Gitea, also he noted some changes with a big comment

Still does not qualify as `provide/link to a clear list of differences` in my opinion. In addition to the examples above (extensive comparison page/tables), here is another good, simpler example: https://github.com/schlagmichdoch/pairdrop#differences-to-snapdrop (if a one-man project can do it, surely forgejo maintainers or contributors can take some time to compile such a list?)

**Comment by nodiscc** [comment]
> "on hold" label for PRs?

In this case `feedback needed` is more appropriate.

**Comment by kokomo123** [comment]
Got it, i created a separate issue after the fact so this doesn't become too cluttered.

**Comment by kokomo123** [comment]
> > he noted some moderation tools that weren't in Gitea, also he noted some changes with a big comment
> 
> Still does not qualify as `provide/link to a clear list of differences` in my opinion. In addition to the examples above (extensive comparison page/tables), here is another good, simpler example: https://github.com/schlagmichdoch/pairdrop#differences-to-snapdrop (if a one-man project can do it, surely forgejo maintainers or contributors can take some time to compile such a list?)

That's fair, i think the maintainers should definitely still create one and i don't know why they haven't made one. I suggested the idea before but I don't know. I'll wait and see what caesar says.

**Comment by caesar** [comment]
@nodiscc I meant to link back to that issue from this PR, but after filling out the PR template I forgot to add it, my bad.
In response to your questions about feature differences from Gitea, little has changed since my last answer but I will summarise here. (Disclosure, as I mentioned there, disclosure, I am a user of and contributor to Forgejo.)

As @kokomo123 points out, _the primary difference is governance_. This is Forgejo's headline feature and the reason the fork was created and why Codeberg supported it and switched to it. I understand from your previous messages that you don't consider governance to be a feature, but it bears repeating because it is the reason for Forgejo's existence.

Beyond that, in the way of concrete functionality, differences are small because _it is a goal of Forgejo to upstream features wherever possible_. It is partly for this reason that there is no official comparison page, because to some extent, significant feature divergence is considered unwanted.

However, there are some feature differences I know of:

- More rigorous [release process](https://forgejo.org/docs/v1.20/developer/release/), which has a few times caught bugs before release that Gitea didn't catch in their releases.
- Different [Actions architecture](https://forgejo.org/docs/v1.20/admin/actions/).
- [Moderation features](https://forgejo.org/docs/v1.20/user/blocking-user/).
- Privacy by default, meaning [a more private update checker mechanism](https://codeberg.org/forgejo-contrib/rvcodns) and some different defaults.
- Different themes.

Unreleased features that are different include:

- [driver](https://codeberg.org/forgejo/forgejo/compare/forgejo-development...forgejo-f3) for the [F3](https://forum.forgefriends.org/t/about-the-friendly-forge-format-f3/681) interchange format

There are probably other differences I'm not thinking of since I'm not one of the core developers, but those are the main ones I know of.

But again, for those involved in Forgejo (which is _extremely_ active) and I believe for most users, [_governance is the main feature_](https://forgejo.org/faq/#why-was-forgejo-created) which differentiates it from Gitea.

**Comment by caesar** [comment]
As for the description (if you do decide to accept this) I am happy to modify it to note the fork relationship however you see fit. I would suggest changing the description to something along the lines of:
> A lightweight software forge focused on scaling, federation, and privacy. Soft fork of Gitea maintained by an inclusive community under the umbrella of the non-profit Codeberg e.V.. Contributes most changes back upstream.

File: software/forgejo.yml
Comment (nodiscc) [inline software/forgejo.yml:4]:
```suggestion
description: A lightweight software forge focused on scaling, federation, and privacy (fork of Gitea).
```

Let's keep it concise and to the point ;)
Agreed with your other comments.
Thanks

**Review by nodiscc — CHANGES_REQUESTED** [review]

File: software/forgejo.yml
Comment (caesar) [reply software/forgejo.yml]: Sure, I've made that change. Thanks @nodiscc .

**Review by nodiscc — APPROVED** [review]

**Review by kokomo123 — APPROVED** [review]
👍🏻 💯

**Pull request merged**
